code/modules/mob/mob.dm

```diff
@@ -105,6 +105,16 @@
 	else
 		to_chat(src, msg)
 
+/mob/proc/display_output(sound/S, image/I, text, turf/turf_source, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+	if(can_hear())
+	//Process sound
+		if(S)
+			SEND_SOUND(src, S)
+	//Process text
+	if(text)
+		to_chat(src, "<span class='italics'>[text]</span>")
```

Comment:
this should be under the previous if clause too, otherwise they will see text "you hear a honk" even if they are deaf

Reply:
i did this cause text might be a visual message, but i guess this is for sound based text only